spec/mem/atomic.md

```diff
@@ -17,9 +27,36 @@ pub enum Atomicity {
     None,
 }
 
+/// Internal type used to track the type of a memory access.
+enum AccessType {
+    Store,
+    Load,
+}
+
+/// Access contains all information the data race detection needs about a single access.
+struct Access {
+    ty: AccessType,
+    atomicity: Atomicity,
+    addr: Address,
+    len: Int,
```

Comment:
I changed it to `Int` because we only use it for the overlapping check and it spared us two `bytes()` and a `Size::from_bytes().unwrap()`. Seems unnecessary.

Reply:
It is conceptually a size though, the size of a memory access. Seems inconsistent to not use the same type as what we have in `load` to indicate memory access sizes.